File: apps/web/shared/types/doc.ts
export interface DocTocLink {
  id: string
  text: string
  depth: number
  children?: DocTocLink[]
}

export interface DocCategoryItem {
  id: number
  slug: string
  title: string
  description: string
  icon: string
  sort_order: number
  created: Date | string
  updated: Date | string
}

// BE returns the shared response.Paginated envelope `{items, total}`
// (apps/api/pkg/response/response.go). FE consumers already read
// `.items` everywhere, so the legacy `categories` / `page` / `limit`
// keys were dead-and-misleading TS.
export interface DocCategoryListResponse {
  items: DocCategoryItem[]
  total: number
}

export interface DocTagItem {
  id: number
  slug: string
  title: string
  description: string
  created: Date | string
  updated: Date | string
}

export interface DocTagListResponse {
  items: DocTagItem[]
  total: number
}

export interface DocArticleCategoryBrief {
  id: number
  slug: string
  title: string
}

export interface DocArticle {
  id: number
  title: string
  slug: string
  path: string
  description: string
  banner: string
  status: number
  isPin: boolean
  view: number
  publishedTime: Date | string
  editedTime: Date | string | null
  contentMarkdown: string
  categoryId: number
  authorId: number
  category: DocArticleCategoryBrief
  // Embedded by BE so the rewrite flow can pre-fill the tag picker
  // without a second fetch. List endpoints may omit; detail always sets.
  tagIds?: number[]
  created: Date | string
  updated: Date | string
  contentHtml?: string
  toc?: DocTocLink[]
}

// Kept for backward compatibility with components that use the old shape
export type DocArticleSummary = DocArticle
export type DocArticleDetail = DocArticle

export interface DocArticleListResponse {
  items: DocArticle[]
  total: number
}

File: apps/api/internal/doc/model/doc.go
package model

import "time"

type DocCategory struct {
	ID          int    `gorm:"primaryKey;autoIncrement" json:"id"`
	Slug        string `gorm:"uniqueIndex;type:varchar(128);not null" json:"slug"`
	Title       string `gorm:"type:varchar(233);not null" json:"title"`
	Description string `gorm:"type:varchar(777);default:''" json:"description"`
	Icon        string `gorm:"type:varchar(128);default:''" json:"icon"`
	SortOrder   int    `gorm:"column:sort_order;default:0" json:"sort_order"`

	CreatedAt time.Time `gorm:"column:created" json:"created"`
	UpdatedAt time.Time `gorm:"column:updated" json:"updated"`
}

func (DocCategory) TableName() string { return "doc_category" }

type DocTag struct {
	ID          int    `gorm:"primaryKey;autoIncrement" json:"id"`
	Slug        string `gorm:"uniqueIndex;type:varchar(128);not null" json:"slug"`
	Title       string `gorm:"type:varchar(128);not null" json:"title"`
	Description string `gorm:"type:varchar(255);default:''" json:"description"`

	CreatedAt time.Time `gorm:"column:created" json:"created"`
	UpdatedAt time.Time `gorm:"column:updated" json:"updated"`
}

func (DocTag) TableName() string { return "doc_tag" }

type DocArticle struct {
	ID              int        `gorm:"primaryKey;autoIncrement" json:"id"`
	Title           string     `gorm:"type:varchar(233);not null" json:"title"`
	Slug            string     `gorm:"uniqueIndex;type:varchar(233);not null" json:"slug"`
	Path            string     `gorm:"uniqueIndex;type:varchar(255);not null" json:"path"`
	Description     string     `gorm:"type:varchar(777);not null" json:"description"`
	Banner          string     `gorm:"type:varchar(777);default:''" json:"banner"`
	Status          int        `gorm:"default:1" json:"status"` // 0=draft, 1=published, 2=archived
	IsPin           bool       `gorm:"column:is_pin;default:false" json:"isPin"`
	View            int        `gorm:"default:0" json:"view"`
	PublishedTime   time.Time  `gorm:"column:published_time;autoCreateTime" json:"publishedTime"`
	EditedTime      *time.Time `gorm:"column:edited_time" json:"editedTime"`
	ContentMarkdown string     `gorm:"column:content_markdown;type:varchar(100007);not null" json:"contentMarkdown"`

	CategoryID int `gorm:"column:category_id;not null" json:"categoryId"`
	AuthorID   int `gorm:"column:author_id;not null" json:"authorId"`

	CreatedAt time.Time `gorm:"column:created" json:"created"`
	UpdatedAt time.Time `gorm:"column:updated" json:"updated"`
}

func (DocArticle) TableName() string { return "doc_article" }

type DocArticleTagRelation struct {
	DocArticleID int `gorm:"column:doc_article_id;primaryKey" json:"doc_article_id"`
	DocTagID     int `gorm:"column:doc_tag_id;primaryKey" json:"doc_tag_id"`

	CreatedAt time.Time `gorm:"column:created" json:"created"`
	UpdatedAt time.Time `gorm:"column:updated" json:"updated"`
}

func (DocArticleTagRelation) TableName() string { return "doc_article_tag_relation" }
